cf-custom-resources/lib/desired-count-delegation.js

```diff
@@ -0,0 +1,131 @@
+ * @returns {Promise} Promise that is resolved on success, or rejected on connection error or HTTP error response
+ */
+let report = function (event, context, responseStatus, physicalResourceId, responseData, reason) {
+    return new Promise((resolve, reject) => {
+        const https = require('https');
+        const {
+            URL
+        } = require('url');
+
+        var responseBody = JSON.stringify({
+            Status: responseStatus,
+            Reason: reason,
+            PhysicalResourceId: physicalResourceId || context.logStreamName,
+            StackId: event.StackId,
+            RequestId: event.RequestId,
+            LogicalResourceId: event.LogicalResourceId,
+            Data: responseData
+        });
+
+        const parsedUrl = new URL(event.ResponseURL || defaultResponseURL);
+        const options = {
+            hostname: parsedUrl.hostname,
+            port: 443,
+            path: parsedUrl.pathname + parsedUrl.search,
+            method: 'PUT',
+            headers: {
+                'Content-Type': '',
+                'Content-Length': responseBody.length
+            }
+        };
+
+        https.request(options)
+            .on('error', reject)
+            .on('response', res => {
+                res.resume();
+                if (res.statusCode >= 400) {
+                    reject(new Error(`Error ${res.statusCode}: ${res.statusMessage}`));
+                } else {
+                    resolve();
+                }
+            })
+            .end(responseBody, 'utf8');
+    });
+};
+
+/**
+ * Get the current running task number for a specific task definition.
+ *
+ * @param {string} cluster Name of the ECS cluster.
+ * @param {string} family Name of the task definition family.
+
+ * @returns {number} The running task number.
+ */
+const getRunningTaskCount = async function (cluster, family) {
+  var ecs = new aws.ECS();
+
+  var taskNum = 0
+  var nextToken;
+  do {
+    const resp = await ecs.listTasks({
+      cluster: cluster,
+      family: family,
+      nextToken: nextToken
+    }).promise();
+    taskNum += resp.taskArns.length;
+    nextToken = resp.nextToken;
+  } while (nextToken)
+
+  return taskNum;
+};
+
+/**
+ * Correct desired count handler, invoked by Lambda.
+ */
+exports.desiredCountHandler = async function(event, context) {
+  var responseData = {};
+  var physicalResourceId;
+
+  try {
+    switch (event.RequestType) {
+      case 'Create':
+        responseData.DesiredCount = event.ResourceProperties.DesiredCount;
+        physicalResourceId = event.PhysicalResourceId;
+        console.log(`Successfully set desired count to ${responseData.DesiredCount}.`);
+        break;
+      case 'Update':
+        responseData.DesiredCount = await getRunningTaskCount(event.ResourceProperties.Cluster, event.ResourceProperties.Family);
+        physicalResourceId = event.PhysicalResourceId;
+        console.log(`Successfully update desired count to ${responseData.DesiredCount}.`);
```

Comment:
I think they are quite useful when debugging the lambda function...and since it doesn't hurt to have them maybe we should keep them?

Reply:
the downside is that customers pay ($0.03 per GB) after the free tier, so I think in our other custom resources we only log in case of an error to bring some visibility but if there are no errors then no logs :)

Reply:
gotcha i'll remove those logs